source/docs/advanced/theme/route-dispatcher.md

```diff
@@ -0,0 +1,134 @@
+---
+id: route-dispatcher
+title: Handle dynamic URLs with the Dispatcher
+---
+
+In some cases, you will need a deeper customization of URLs than what you can do in the [Add a new page guide](../../essentials/add-a-page-client-side). For instance, for some SEO reasons, you may prefer to have `/my-product` instead of `/product/my-product-slug`.
+
+That is what the Dispatcher is for in Front-Commerce, and what we will cover in this documentation.
+
+## What is the goal of the Dispatcher?
+
+The dispatcher is actually a component within Front-Commerce that will be displayed in case no other route was found. Its goal will be to ask the server (by using the `matchUrls` query in your GraphQL Schema) what kind of page is associated with the current URL and will display the page's component accordingly.
+
+<figure>
+![Diagram explaining how an URL is displayed](./assets/dispatcher.svg)
+</figure>
+
+If you come from a Magento background, this is the concept behind [URL Rewrites](https://docs.magento.com/m2/ce/user_guide/marketing/url-rewrite.html).
+
+In the core of Front-Commerce, the association between an URL and a page is already done for things like Products, Categories, CMS pages… But depending on your own site, you might need to add new ones.
+
+To do so, you will need to proceed in two steps:
+* Support the new URLs in the `matchUrls` query in your GraphQL Schema
+* Add the mapping between the type returned by `matchUrls` and the page component that should be displayed
+
+## Support the new URLs in the `matchUrls`
+
+<blockquote class="info">
+**Magento:** If the url you are trying to add are managed by Magento, you don't need any of this. You should instead [add an url rewrite directly in your backend](https://devdocs.magento.com/guides/v2.2/cloud/configure/import-url-rewrites.html), since the mechanism already exists in the Magento module of Front-Commerce.
+</blockquote>
+
+The goal here will be [to override the resolver](/docs/reference/graphql-module-definition.html#resolvers-optional) of `matchUrls` in order to add your own URLs.
+
+1. Create a custom GraphQL module that will add new `resolvers` to your GraphQL Schema See [Create a new GraphQL module](/docs/essentials/extend-the-graphql-schema.html) for more details.
+2. Create a loader that will match a URL string with an Entity (the object returned below):
+```js
+const MyModuleUrlLoader = makeDataLoader => () => {
+  return {
+    load: url => {
+      if (url === "my-dynamic-custom-url") {
+        return {
+          // The url that you are currently matching
+          url: "my-dynamic-custom-url",
+          // The type of entity you are willing to
+          // display for this URL
+          type: "pageType",
```

Comment:
For more consistency, I suggest we use `myPageType` instead in the whole page.

It makes more explicit that this is an example and not an existing type or something like that.